hail/src/main/scala/is/hail/expr/ir/functions/UtilFunctions.scala

```diff
@@ -48,6 +48,25 @@ object UtilFunctions extends RegistryFunctions {
     case _ => s.toDouble
   }
 
+  def isValidBoolean(s: String): Boolean =
+    (s equalsCI "true") || (s equalsCI "false")
+
+  def isValidInt32(s: String): Boolean =
+    try { s.toInt; true } catch { case _: NumberFormatException => false }
+
+  def isValidInt64(s: String): Boolean =
+    try { s.toLong; true } catch { case _: NumberFormatException => false }
+
+  def isValidFloat32(s: String): Boolean = parseSpecialNum(s) match {
+    case 0 => try { s.toFloat; true } catch { case _: NumberFormatException => false }
+    case _ => true
+  }
+
+  def isValidFloat64(s: String): Boolean = parseSpecialNum(s) match {
```

Comment:
You're parsing twice here -- once for the check and once for the parse. That's not ideal, but I do see the reason why it's not easy to do it just once in generated code without boxing.

Reply:
yeah :I if we could catch exceptions in asm4s then that would be one nicer way to handle this.